### ISSUE TITLE
Dev Question: Want to use an ESP8266 (Nodemcu) as usb interface

### ISSUE THREAD

**Issue by nospam2k** [mcallegari/qlcplus#1332]
First, I want to say I really appreciate your time and effort in developing such a full featured Light Controller!

I am fairly new to DMX so I may have the wrong premise but:

I am successfully using QLC+ with artnet, but I'm not happy about sending unneeded udp data over my network. I had previously started to write my own mini-controller software that works fine for just dimming lights before finding QLC+. 

I want to use QLC+ with the usb on a nodemcu to just send packets to my other device.

My problem is the dmx usb plugin needs ftdi. I have seen some mention of a uart plugin. So, I'm wanting to know is there a generic usb interface plugin available? If not, what is the quickest way to recompile the dmx usb (ubuntu) without recompiling all qlc+? I can then use that as a base for the 8266.

Thanks again for such a great piece of software!

**Comment by nospam2k** [comment]
I have compiled QLC+ from scratch and no problems (accept for qt deprecations which I just replaced -Werror with -Wno-deprecations-declaration) so I'm ok with just doing a full compile as it's fast. I tried just adding the vender id and product id to dmxinterface.h and it recognized the interface but isn't even trying to send data. On the ESP8266 I just have it echoing between Serial and Serial1. I found I had to sudo qlcplus in Xubuntu so I'm sure there is some user I need to add to some group, but it shows. The serial port says it's a CH430 so I imagine it isn't an exact comparison to FTDI but not sure where to go from here. Thanks for any help. I readily admit I know enough to be dangerous but no enough to get this working so I really appreciate any thoughts on this.
